sympy/solvers/solveset.py

```diff
@@ -287,14 +287,14 @@ def _invert_real(f, g_ys, symbol):
                 if b:
                     return _invert_real(expo, FiniteSet(s), symbol)
                 else:
-                    return _invert_real(expo, S.EmptySet, symbol)
+                    return (expo, S.EmptySet)
```

Comment:
I would leave this and the next change alone -- let it recurse so you only have to handle empty set in one place.

Reply:
But it is pointless solving over S.EmptySet. Slightly better now with the early exit, but why even go down that path?

Reply:
Because it's possible (though I haven't checked) that the caller will look at the first arg to see if the expression has been solved, i.e. that the symbol has been isolated. By returning the expression yo circumvent handling for that case.